protocol/verification.go

```diff
@@ -0,0 +1,51 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
```

Comment:
I don't think we have a function yet for the client to verify that the `userLeafNode` value (i.e. the key and user policy flags) haven't changed since we last checked. This is especially important for monitoring (and while we don't have key changes) to ensure that the server hasn't inserted a valid, yet fake, key for the name.